### ISSUE TITLE
【PaddlePaddle Hackathon 2】17、为 Paddle 新增 paddle.nn.CosineEmbeddingLoss 和 paddle.nn.functional.cosine_embedding_loss API

### ISSUE THREAD

**Issue by TCChenlong** [PaddlePaddle/Paddle#40316]
（此 ISSUE 为 PaddlePaddle Hackathon 第二期活动的任务 ISSUE，更多详见 [【PaddlePaddle Hackathon 第二期】任务总览](https://github.com/PaddlePaddle/Paddle/issues/40234)）

【任务说明】
任务标题：为 Paddle 新增 paddle.nn.CosineEmbeddingLoss 和 paddle.nn.functional.cosine_embedding_loss API.

技术标签：深度学习框架，Python

任务难度：简单

详细描述：CosineEmbeddingLoss, 即余弦相似度损失函数，用于衡量两个向量的相似程度，常用于非线性嵌入学习、半监督学习等领域。该任务目标为向paddle中添加计算该损失函数的功能，并提供类实例paddle.nn.CosineEmbeddingLoss和函数paddle.nn.functional.cosine_embedding_loss两种调用方式。

【提交流程】

请按 [新增API 开发&提交流程](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/dev_guides/api_contributing_guides/api_contributing_guides_cn.html)中的描述，完成任务提交过程中的相关步骤。

【提交内容】

- API 的设计文档，并提 PR 至 [PaddlePaddle/community](https://github.com/PaddlePaddle/community) 的 rfcs/APIs 目录。
- Python 实现代码 & 英文 API 文档，在 Paddle repo 的 [python/paddle/nn/functional/loss.py](https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/nn/functional/loss.py)实现函数调用，[python/paddle/nn/layer/loss.py](https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/nn/layer/loss.py)中实现类实例调用。新增接口初始化在[python/paddle/nn/functional/__init__.py](https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/nn/functional/__init__.py)与 [paddle/nn/layer/__init__.py](https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/nn/layer/__init__.py) 。
- 单测代码，在 Paddle repo 的 [python/paddle/fluid/tests/unittests](https://github.com/PaddlePaddle/Paddle/tree/develop/python/paddle/fluid/tests/unittests) 目录
- 中文API文档，在 docs repo 的  docs/api/paddle/nn 目录下新建 CosineEmbeddingLoss_cn.rst， [docs/api/paddle/nn/functional](https://github.com/PaddlePaddle/docs/tree/develop/docs/api/paddle/nn/functional) 目录下新建 cosine_embedding_loss_cn.rst。

【合入标准】

- 1、按 [API 设计规范](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/dev_guides/api_contributing_guides/api_design_guidelines_standard_cn.html) 完成 API设计文档；
- 2、按[API 验收标准](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/dev_guides/api_contributing_guides/api_accpetance_criteria_cn.html)完成 API功能实现、单测、API文档；

【技术要求】

- 熟悉 CosineEmbeddingLoss 的数学原理和适用场景
- 熟悉 Paddle 动静态图下数学计算过程
- 熟练掌握Python

【参考内容】

- [文档：API 贡献指南](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/dev_guides/api_contributing_guides/api_contributing_guides_cn.html)

【答疑交流】

- 如果在开发中对于上述任务有任何问题，欢迎在本 ISSUE 下留言交流。
- 对于开发中的共性问题，在活动过程中，会定期组织答疑，请大家关注官网&QQ群的通知，及时参与。
